recipes/arm-gnu-toolchain/all/conanfile.py

```diff
@@ -81,6 +81,18 @@ def validate(self):
                 f"{supported_build_architectures[build_os]}."
             )
 
+        compiler_name = str(self.settings.compiler)
+        if compiler_name != "gcc":
+            raise ConanInvalidConfiguration(
+                "The compiler must be set to gcc to use this toolchain.")
+
+        compiler_version = str(self.settings.compiler.version)
+        supported_compiler_versions = ["11.3", "12.2", "12.3"]
```

Comment:
The consumers can also specify a major version of the compiler, i.e. `11` or `12`, to allow broader binary compatibility. It would be good to allow those cases as well.

```suggestion
        supported_compiler_versions = ["11", "11.3", "12", "12.2", "12.3"]
```